src/Gateways/GpApiGateway.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace GlobalPayments\WooCommercePaymentGatewayProvider\Gateways;
+
+use GlobalPayments\Api\Entities\Enums\Environment;
+use GlobalPayments\Api\Entities\Enums\GatewayProvider;
+
+defined( 'ABSPATH' ) || exit;
+
+class GpApiGateway extends AbstractGateway {
+	/**
+	 * SDK gateway provider
+	 *
+	 * @var string
+	 */
+	public $gateway_provider = GatewayProvider::GP_API;
+
+	/**
+	 * App ID
+	 *
+	 * @var string
+	 */
+	public $app_id;
+
+	/**
+	 * App Key
+	 *
+	 *
+	 * @var string
+	 */
+	public $app_key;
+
+	/**
+	 * Should live payments be accepted
+	 *
+	 * @var bool
+	 */
+	public $is_production;
+
+	/**
+	 * Integration's Developer ID
+	 *
+	 * @var string
+	 */
+	public $developer_id = '';
+
+	public function configure_method_settings() {
+		$this->id                 = 'globalpayments_gpapi';
+		$this->method_title       = __( 'GP-API', 'globalpayments-gateway-provider-for-woocommerce' );
+		$this->method_description = __( 'Connect to the Global Payments API (GP-API) gateway', 'globalpayments-gateway-provider-for-woocommerce' );
+	}
+
+	public function get_first_line_support_email() {
+		return 'api.integrations@globalpay.com';
+	}
+
+	public function get_gateway_form_fields() {
+		return array(
+			'app_id' => array(
+				'title'       => __( 'App Id', 'globalpayments-gateway-provider-for-woocommerce' ),
+				'type'        => 'text',
+				'description' => __(
+					'Get your App Id from your <a href="https://developer.globalpay.com/user/register" target="_blank">Global Payments Developer Account</a>.',
+					'globalpayments-gateway-provider-for-woocommerce'
+				),
+				'default'     => '',
+			),
+			'app_key' => array(
+				'title'       => __( 'App Key', 'globalpayments-gateway-provider-for-woocommerce' ),
+				'type'        => 'text',
+				'description' => __(
+					'Get your App Key from your <a href="https://developer.globalpay.com/user/register" target="_blank">Global Payments Developer Account</a>.',
+					'globalpayments-gateway-provider-for-woocommerce'
+				),
+				'default'     => '',
+			),
+			'is_production' => array(
+				'title'   => __( 'Live Mode', 'globalpayments-gateway-provider-for-woocommerce' ),
+				'type'    => 'checkbox',
+				'default' => 'no',
+			),
+		);
+	}
+
+	public function get_frontend_gateway_options() {
+		return array(
+			'accessToken' => $this->get_access_token(),
```

Comment:
Something that just came to mind. We need to use access tokens that only have single-use tokenization privileges on the client side. This will be to prevent access tokens with higher privileges from being leaked.